### PR TITLE
Redux basic flow

### PR DESCRIPTION
**Pull request by jimmyfnugent**
Same basic functionality, but this is now running through Redux state, with actions and reducers and a store!

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import makeStore from './src/store';
+
+export const store = makeStore();
```
Comment (hopskipnfall) [inline server/index.js]: NEWLINE AT END IS SUPER IMPORTANT

File: server/test/reducer_test.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import {expect} from 'chai';
 import {Map, fromJS} from 'immutable';
 
-import {addBit} from '../src/bits_control';
+import reducer from '../src/reducer';
 
-describe('bits control logic', () => {
+describe('reducer logic: ', () => {
```
Comment (hopskipnfall) [inline server/test/reducer_test.js]: no colon i think

File: server/test/reducer_test.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import {expect} from 'chai';
 import {Map, fromJS} from 'immutable';
 
-import {addBit} from '../src/bits_control';
+import reducer from '../src/reducer';
 
-describe('bits control logic', () => {
+describe('reducer logic: ', () => {
   describe('add bit', () => {
     it('adds a bit to an empty initial state', () => {
```
Comment (hopskipnfall) [inline server/test/reducer_test.js]: should add

File: server/test/reducer_test.js [hunk 4]
```diff
@@ -17,7 +17,8 @@ describe('bits control logic', () => {
         title: 'Fox is unedgeguardable',
         content: 'No matter what you do, you\'ll never be able to kill a recovering Fox.',
       });
-      const newState = addBit(Map(), bit);
+      const action = {type: 'ADD_BIT', data: bit};
+      const newState = reducer(undefined, action);
```
Comment (hopskipnfall) [inline server/test/reducer_test.js]: this looks ridiculous. i've never seen anyone actually *use* `undefined` before.

File: server/index.js (hunk 1 above)
Comment (jimmyfnugent) [reply server/index.js]: Done

File: server/test/reducer_test.js (hunk 4 above)
Comment (jimmyfnugent) [reply server/test/reducer_test.js]: Well, it's supposed to be if there's no initial state... if there's a better way to do this then I'm all ears :-)

File: server/test/reducer_test.js (hunk 3 above)
Comment (jimmyfnugent) [reply server/test/reducer_test.js]: Done

File: server/test/reducer_test.js (hunk 2 above)
Comment (jimmyfnugent) [reply server/test/reducer_test.js]: Done

**Pull request merged**